ecosystem/indexer-grpc/indexer-grpc-post-processor/src/file_storage_verifier.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright © Aptos Foundation
+
+use crate::metrics::VERIFICATION_ERROR_COUNT;
+use anyhow::{bail, ensure};
+use aptos_indexer_grpc_utils::{
+    config::IndexerGrpcFileStoreConfig,
+    constants::BLOB_STORAGE_SIZE,
+    file_store_operator::{
+        FileStoreOperator, GcsFileStoreOperator, LocalFileStoreOperator, TransactionsFile,
+    },
+};
+use aptos_protos::transaction::v1::Transaction;
+use prost::Message;
+
+pub struct FileStorageVerifier {
+    pub file_store_config: IndexerGrpcFileStoreConfig,
+    pub chain_id: u64,
+}
+
+impl FileStorageVerifier {
+    pub fn new(file_store_config: IndexerGrpcFileStoreConfig, chain_id: u64) -> Self {
+        Self {
+            file_store_config,
+            chain_id,
+        }
+    }
+
+    pub async fn run(&self) -> anyhow::Result<()> {
+        let mut file_store_operator: Box<dyn FileStoreOperator> = match &self.file_store_config {
+            IndexerGrpcFileStoreConfig::GcsFileStore(gcs_file_store) => {
+                Box::new(GcsFileStoreOperator::new(
+                    gcs_file_store.gcs_file_store_bucket_name.clone(),
+                    gcs_file_store
+                        .gcs_file_store_service_account_key_path
+                        .clone(),
+                ))
+            },
+            IndexerGrpcFileStoreConfig::LocalFileStore(local_file_store) => Box::new(
+                LocalFileStoreOperator::new(local_file_store.local_file_store_path.clone()),
+            ),
+        };
+        // Verify the existence of the storage bucket.
+        file_store_operator.verify_storage_bucket_existence().await;
+        // Get or create verification metadata file.
+        let verification_metadata = file_store_operator
+            .get_or_create_verification_metadata(self.chain_id)
+            .await?;
+        let file_store_metadata = file_store_operator
+            .get_file_store_metadata()
+            .await
+            .ok_or(anyhow::anyhow!("File Store metadata does not exist"))?;
+        ensure!(
+            file_store_metadata.chain_id == self.chain_id,
+            "Chain ID mismatch"
+        );
+        let mut next_version_to_verify = verification_metadata.next_version_to_verify;
+        let mut current_head_version = file_store_metadata.version;
+
+        loop {
+            if next_version_to_verify > current_head_version {
```

Comment:
Since `next_version_to_verify` gets incremented in line 111, won't this condition sometimes be true?

Reply:
good call; change it to `next_version_to_store` from `current_head_version`.

i should improve my skill of naming things 🤦